sktime/forecasting/trend/_function_forecaster.py

```diff
@@ -0,0 +1,133 @@
+# !/usr/bin/env python3 -u
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implements FunctionForecaster."""
+
+__author__ = ["benheid"]
+__all__ = ["FunctionForecaster"]
+
+import numpy as np
+import pandas as pd
+from scipy.optimize import curve_fit
+
+from sktime.forecasting.base import BaseForecaster, ForecastingHorizon
+
+
+class FunctionForecaster(BaseForecaster):
+    """The FunctionForecaster uses a function that is fitted using scipy curve_fit.
+
+    Parameters
+    ----------
+    function: Callable
+        The function that should be fitted and used to make forecasts.
+    initial_params : list
+        The initial parameters of the functions that should be used for fitting
+
+
+    Examples
+    --------
+    >>> from sktime.forecasting.trend import FunctionForecaster
+    >>> from sktime.datasets import load_airline
+    >>> y = load_airline()
+    >>> def linear_function(x, a, b):
+    ...     return a * x + b
+    >>> forecaster = FunctionForecaster(function=linear_function,
+    ...                                 initial_params=[1, 1])
+    >>> forecaster.fit(y)
+    FunctionForecaster(...)
+    >>> y_pred = forecaster.predict(fh=[1, 2, 3])
+    """
+
+    _tags = {
+        "scitype:y": "univariate",
+        "y_inner_mtype": "pd.Series",
+        "X_inner_mtype": "pd.DataFrame",
+        "ignores-exogeneous-X": True,
+        "requires-fh-in-fit": False,
+    }
+
+    def __init__(self, function, initial_params):
+        self.function = function
+        self.initial_params = initial_params
```

Comment:
In the context of forecasting, how important is `initial_params` for it to be marked as a non-default, non-None parameter? Scipy provides the provision to fill the initial values (`p0`) with 1s by inspecting the function. Perhaps we can make use of that to let the user only provide the function?

Reply:
Good idea :)

Reply:
I make all parameters except for `full_output` available to the user.

Reply:
I need to pass them via a dict to avoid interactions with different scipy versions..